server.go

```diff
@@ -243,6 +245,36 @@ func (s *Server) register(sd *ServiceDesc, ss interface{}) {
 	s.m[sd.ServiceName] = srv
 }
 
+// ServiceMetadata returns the metadata for a service or method.
+// service should be the full service name with package, in the form of <package>.<service>.
+// method should be the method name only.
+// If only service is important, method should be an empty string.
+func (s *Server) ServiceMetadata(service, method string) interface{} {
```

Comment:
The way you're describing this function, I'm curious why it is an exported function?

Do we ever expect importers to call this function? Should they ever call it? Is the only real thing an importer should be caring about is if return value == nil?

If the later, wouldn't it be better to use an exported method like IsValidService, or IsSupportedService, etc?